clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java

```diff
@@ -108,17 +121,22 @@ public boolean shouldRetryRequest(final Throwable throwable) {
   private void loadCredentials() throws IOException {
     Optional<ZeebeClientCredentials> cachedCredentials;
 
+    cacheLock.lock();
```

Comment:
❓ Do we need this lock if we've already made the cache thread-safe?

Reply:
Hey @remcowesterhoud!

Yes, you made a good point. Let me rethink it a little bit.

Reply:
@remcowesterhoud Can you please take another look? I refactored it a bit.